src/ppom-controller.ts

```diff
@@ -418,8 +418,12 @@ export class PPOMController extends BaseControllerV2<
    * to update the configuration.
    */
   async #maybeUpdatePPOM(): Promise<void> {
-    if (this.#isDataRequiredForCurrentChain()) {
+    const versionInfoUpdated = await this.#updateVersionInfo();
+    if (versionInfoUpdated || this.#isDataRequiredForCurrentChain()) {
```

Comment:
If I understand correctly, what's happening here is when a transaction is initialized, we check if there's any updates available before proceeding.

AFAIR the idea was to check for updates availability often enough so that we can keep relying on the information we have instead of looking up the update status.
Getting the update is not revealing which chain is used but it is revealing that the user is going to be looking at a transaction in a moment. 

Is this change fixing the time gap in which an update from CDN is not noticed? We were supposed to have an acceptable period of that. Why is it considered a bug?

Reply:
I changed the approach in this PR. The bug reported by blockaid team is bit of edge case can happen only if CDN is updated and new data is not yet fetched.

If CDN is updated we can not depend on old version info as those files may no longer be available in CDN.

The new approach reset dataFetched property for network if CDN is updated and thus only new files added will be fetched from CDN before the transaction.

Reply:
I created this new task to address the issue: https://app.zenhub.com/workspaces/-confirmations-secure-ux-6245e6e2348677001213b8d2/issues/gh/metamask/metamask-extension/new?issueType=issue